tools/csv-merger/csv-merger.go

```diff
@@ -0,0 +1,375 @@
+	if len(overlapsMap) != 0 {
+		log.Print("ERROR: Overlapping API Groups were found between different operators.")
+		for apigroup := range overlapsMap {
+			fmt.Print("The API Group " + apigroup + " is being used by these operators: " + strings.Join(overlapsMap[apigroup], ", ") + "\n")
+			return false
+		}
+	}
+	return true
+}
+
+func main() {
+	flag.Parse()
+
+	if *crdDir != "" {
+		result := validateNoApiOverlap(*crdDir)
+		if result {
+			os.Exit(0)
+		} else {
+			os.Exit(1)
+		}
+	}
+
+	switch *outputMode {
+	case CSVMode:
+		if *specDisplayName == "" || *specDescription == "" {
+			panic(errors.New("Must specify spec-displayname and spec-description"))
+		}
+
+		csvs := []string{
+			*novaCsv,
+			*neutronCsv,
+			*computeNodeCsv,
+		}
+
+		version := semver.MustParse(*csvVersion)
+		var replaces string
+		if *replacesCsvVersion != "" {
+			replaces = fmt.Sprintf("%v.v%v", operatorName, semver.MustParse(*replacesCsvVersion).String())
+		}
+
+		// This is the basic CSV without an InstallStrategy defined
+		csvBase := operator.GetCSVBase(
+			operatorName,
+			*namespace,
+			*specDisplayName,
+			*specDescription,
+			*operatorImage,
+			replaces,
+			version,
+			*crdDisplay,
+		)
+		csvExtended := ClusterServiceVersionExtended{
+			TypeMeta:   csvBase.TypeMeta,
+			ObjectMeta: csvBase.ObjectMeta,
+			Spec:       ClusterServiceVersionSpecExtended{ClusterServiceVersionSpec: csvBase.Spec},
+			Status:     csvBase.Status}
+
+		// This is the base deployment + rbac for the OpenStack Cluster CSV
+		installStrategyBase := operator.GetInstallStrategyBase(
+			*namespace,
+			*operatorImage,
+			"IfNotPresent",
+		)
+
+		for _, image := range strings.Split(*relatedImagesList, ",") {
+			if image != "" {
+				name := ""
+				if strings.Contains(image, "|") {
+					image_s := strings.Split(image, "|")
+					image = image_s[0]
+					name = image_s[1]
+				} else {
+					names := strings.Split(strings.Split(image, "@")[0], "/")
+					name = names[len(names)-1]
+				}
+				csvExtended.Spec.RelatedImages = append(
+					csvExtended.Spec.RelatedImages,
+					relatedImage{
+						Name: name,
+						Ref:  image,
+					})
+			}
+		}
+
+		for _, csvStr := range csvs {
+			if csvStr != "" {
+				csvBytes := []byte(csvStr)
+
+				csvStruct := &csvv1alpha1.ClusterServiceVersion{}
+
+				err := yaml.Unmarshal(csvBytes, csvStruct)
+				if err != nil {
+					panic(err)
+				}
+
+				installStrategyBase.DeploymentSpecs = append(installStrategyBase.DeploymentSpecs, csvStruct.Spec.InstallStrategy.StrategySpec.DeploymentSpecs...)
+				installStrategyBase.ClusterPermissions = append(installStrategyBase.ClusterPermissions, csvStruct.Spec.InstallStrategy.StrategySpec.ClusterPermissions...)
+				installStrategyBase.Permissions = append(installStrategyBase.Permissions, csvStruct.Spec.InstallStrategy.StrategySpec.Permissions...)
+
+				for _, owned := range csvStruct.Spec.CustomResourceDefinitions.Owned {
+					csvExtended.Spec.CustomResourceDefinitions.Owned = append(
```

Comment:
I'm pretty sure this is safe, according to my understanding of slices. The append will create a new slice every time it's called, and as we only call append we won't affect any of the underlying data being iterated over by range.